actors/builtin/storage_power/policy.go

```diff
@@ -1,66 +1,89 @@
+	// PARAM_FINISH: always penalise the entire pledge.
 	switch faultType {
-	case DoubleForkMiningFault:
-		return currPledge
-	case ParentGrindingFault:
-		return currPledge
-	case TimeOffsetMiningFault:
-		return currPledge
+	case ConsensusFaultDoubleForkMining:
+		return pledge
+	case ConsensusFaultParentGrinding:
+		return pledge
+	case ConsensusFaultTimeOffsetMining:
+		return pledge
 	default:
 		panic("Unsupported case for pledge collateral consensus fault slashing")
 	}
 }
 
-func rewardForConsensusSlashReport(elapsedEpoch abi.ChainEpoch, collateralToSlash abi.TokenAmount) abi.TokenAmount {
+var consensusFaultReporterInitialShare = BigFrac{
+	// PARAM_FINISH
+	numerator:   big.NewInt(1),
+	denominator: big.NewInt(1000),
+}
+var consensusFaultReporterShareGrowthRate = BigFrac{
+	// PARAM_FINISH
+	numerator:   big.NewInt(102813),
+	denominator: big.NewInt(100000),
+}
+
+func rewardForConsensusSlashReport(elapsedEpoch abi.ChainEpoch, collateral abi.TokenAmount) abi.TokenAmount {
 	// PARAM_FINISH
 	// var growthRate = SLASHER_SHARE_GROWTH_RATE_NUM / SLASHER_SHARE_GROWTH_RATE_DENOM
 	// var multiplier = growthRate^elapsedEpoch
 	// var slasherProportion = min(INITIAL_SLASHER_SHARE * multiplier, 1.0)
-	// return collateralToSlash * slasherProportion
+	// return collateral * slasherProportion
 
 	// BigInt Operation
-	// NUM = SLASHER_SHARE_GROWTH_RATE_NUM^elapsedEpoch * INITIAL_SLASHER_SHARE_NUM * collateralToSlash
+	// NUM = SLASHER_SHARE_GROWTH_RATE_NUM^elapsedEpoch * INITIAL_SLASHER_SHARE_NUM * collateral
 	// DENOM = SLASHER_SHARE_GROWTH_RATE_DENOM^elapsedEpoch * INITIAL_SLASHER_SHARE_DENOM
-	// slasher_amount = min(NUM/DENOM, collateralToSlash)
-	slasherShareNumMultiplier := big.NewInt(1)
-	for i := int64(0); i < int64(elapsedEpoch); i++ {
-		slasherShareNumMultiplier = big.Mul(slasherShareNumMultiplier, indices.ConsensusFault_SlasherShareGrowthRateNum())
-	}
+	// slasher_amount = min(NUM/DENOM, collateral)
+	elapsed := big.NewInt(int64(elapsedEpoch))
+	slasherShareNumerator := big.Exp(consensusFaultReporterShareGrowthRate.numerator, elapsed)
+	slasherShareDenominator := big.Exp(consensusFaultReporterShareGrowthRate.denominator, elapsed)
 
-	slasherShareDenomMultiplier := big.NewInt(1)
-	for i := int64(0); i < int64(elapsedEpoch); i++ {
-		slasherShareDenomMultiplier = big.Mul(slasherShareDenomMultiplier, indices.ConsensusFault_SlasherShareGrowthRateDenom())
-	}
-
-	num := big.Mul(big.Mul(slasherShareNumMultiplier, indices.ConsensusFault_SlasherInitialShareNum()), collateralToSlash)
-	denom := big.Mul(slasherShareDenomMultiplier, indices.ConsensusFault_SlasherInitialShareDenom())
-	return big.Min(big.Div(num, denom), collateralToSlash)
+	num := big.Mul(big.Mul(slasherShareNumerator, consensusFaultReporterInitialShare.numerator), collateral)
+	denom := big.Mul(slasherShareDenominator, consensusFaultReporterInitialShare.denominator)
+	return big.Min(big.Div(num, denom), collateral)
```

Comment:
I'm not quite sure an exponential is the right choice for the reporter's share - either this will be vanishingly small at some points, or else will saturate to 1 at some points. Could we leave the entire function parametric?

Reply:
No, a concrete function is required. This PR is not about changing it. When there's a new mechanism proposal, we can implement it.

Reply:
As a note, in lotus we implemented this as a lookup table.